tests/Progress/SnapshotTest.php

```diff
@@ -359,12 +361,223 @@ public function of_activity_runs_the_above_for_all_activity_instances()
         $activityInstance2 = factory(ActivityInstance::class)->create(['activity_id' => $activity->id]);
         $activityInstance3 = factory(ActivityInstance::class)->create(['activity_id' => $activity->id]);
 
-        $progresses = (new Snapshot())->ofActivity($activity);
+        $progresses = (new Snapshot(new ProgressUpdateRepository()))->ofActivity($activity);
         $this->assertEquals($activityInstance1->id, $progresses[0]->getActivityInstanceId());
         $this->assertEquals($activityInstance2->id, $progresses[1]->getActivityInstanceId());
+
         $this->assertEquals($activityInstance3->id, $progresses[2]->getActivityInstanceId());
         $this->assertEquals($activity->id, $progresses[0]->getActivityId());
         $this->assertEquals($activity->id, $progresses[1]->getActivityId());
         $this->assertEquals($activity->id, $progresses[2]->getActivityId());
     }
+
+    public function createModuleInstanceProgress(int $module, bool $active, bool $visible, bool $complete, bool $mandatory, int $percentage): ModuleInstanceProgress
+    {
+        $moduleEvaluation = new ModuleInstanceProgress();
+        $moduleEvaluation->setModuleInstanceId($module);
+        $moduleEvaluation->setActive($active);
+        $moduleEvaluation->setVisible($visible);
+        $moduleEvaluation->setComplete($complete);
+        $moduleEvaluation->setMandatory($mandatory);
+        $moduleEvaluation->setPercentage($percentage);
+        return $moduleEvaluation;
+    }
+
+    public function createModuleEvaluationInstance(bool $active, bool $visible, bool $complete, bool $mandatory, int $percentage): Evaluation
+    {
+        $moduleEvaluation = new Evaluation();
```

Comment:
Fyi: usually, we should mock the evaluation class because we shouldn't be testing it here. In this case, however, the Evaluation class is so simple (it just holds data) that it's fine to new up an instance